org.eclipse.jdt.core/model/org/eclipse/jdt/internal/core/CompilationUnit.java

```diff
@@ -783,7 +783,7 @@ private static boolean matchSignatures(IMethodBinding invocation, IMethodBinding
 
 
 private org.eclipse.jdt.core.dom.CompilationUnit getOrBuildAST(WorkingCopyOwner workingCopyOwner) {
-	if (this.ast == null) {
+	if (this.ast == null || this.isWorkingCopy()) {
```

Comment:
That's cool if it works, but then it means we will compute the AST multiple time even if the working copy is not modified. Isn't there a way to detect whether it's worth reparsing? Can `isConsistent` help here?

Reply:
No, it can't.  I trying to use `hasUnsavedChanges`.

Reply:
I have updated the PR.

Reply:
I still think we can find some better condition that this as we would like to avoid that calling twice this method on the same unsaved working copy does re-create an AST every time. But at the moment, correctness is much more important that performance and again I'm optimistic we can find something that works some time later.
Can you please just add a note as a comment in the code to ring a bell about this concern when it's time to consider it?

Reply:
> Now, we check if the copy has unsaved changes 

Yes, but this can return `true` multiple time, and thus trigger multiple AST creations, even if the content of the (unsaved) working copy wasn't modified between. For example, if you have a dirty file, and call hover on multiple locations; then the same AST should ideally be reused; this patch will re-create one everytime.

> Could you, please, suggest what you want me to add?

You can mostly copy what I've put in my former comment.

Reply:
> You can mostly copy what I've put in my former comment.

I have updated the PR.